src/Interpreters/ConcurrentHashJoin.cpp

```diff
@@ -327,7 +327,7 @@ bool ConcurrentHashJoin::addBlockToJoin(const Block & right_block_, bool check_l
         }
     }
 
-    if (check_limits)
+    if (check_limits && table_join->sizeLimits().hasLimits())
```

Comment:
It might not be obvious per se. We do want to avoid calls to `getTotalRowCount` and `getTotalByteCount` because, within them, we will iterate over all inner `HashJoin` objects in turn and lock their mutexes. It creates significant contention and lowers the `addBlockToJoin` parallelism.

Reply:
How it looked:

<img width="1247" alt="Screenshot 2025-01-28 at 19 42 42" src="https://github.com/user-attachments/assets/6dc06fa3-b4de-45ba-8376-3609aee059d9" />

How it should look:

<img width="1247" alt="Screenshot 2025-01-28 at 19 40 20" src="https://github.com/user-attachments/assets/e59a4a3a-10c0-4797-bf95-e98ee2433fed" />